// @netlify/sdk--ui-functions/dist/trpc/main.js
import {} from "@netlify/functions";
import {} from "@trpc/server";
import { fetchRequestHandler } from "@trpc/server/adapters/fetch";
import { createNetlifySDKContext, } from "../sdk_context.js";
export const createNetlifyTRPCHandler = ({ router, endpoint, createContext, }) => {
    return async (req, netlifyContext) => {
        return fetchRequestHandler({
            // TODO(ndhoule): Honestly, this is a little funky. Maybe it's better to pass a context
            // customizer function that is expected to return the full context passed to trpc?
            createContext: (options) => {
                const netlifySDKContext = createNetlifySDKContext({
                    req: options.req,
                    context: netlifyContext,
                });
                const customContext = createContext?.({
                    ...options,
                    context: { ...netlifyContext, ...netlifySDKContext },
                }) ?? {};
                return {
                    ...netlifyContext,
                    ...netlifySDKContext,
                    ...customContext,
                };
            },
            // TODO(ndhoule): Is it possible to eliminate the need for integrators to pass this? (Probably no.)
            endpoint,
            onError: (opts) => {
                console.error("TRPC request failed", { error: opts.error });
            },
            req,
            router,
        });
    };
};
