// @netlify/sdk--ui-core/dist/message_schema/host/initialize.js
import { Surfaces } from "../../surfaces.js";
/*
 * The INITIALIZE message is sent when the host has received an initialization event from the
 * surface. It should contain most, if not all, data the surface would need to start begin rendering
 * (e.g. metadata about the extension, authentication credentials, etc.)
 */
import { z } from "zod";
export const UI_EXTENSION_HOST_INITIALIZE_EVENT_TYPE = "NF::EUI_HOST::INITIALIZE";
export const UIExtensionHostInitializeEventSchema = z.object({
    type: z.literal(UI_EXTENSION_HOST_INITIALIZE_EVENT_TYPE),
    data: z.object({
        context: z.object({
            auth: z.object({
                netlifyToken: z.string().nullable().default(null),
                providerToken: z.string().nullable().default(null),
                providerTokens: z
                    .array(z.object({
                    externalId: z.string().min(1).trim(),
                    token: z.string().min(1).trim(),
                    connectionId: z.string().min(1).trim(),
                }))
                    .default(() => []),
            }),
            configurationId: z.string().nullable().default(null),
            dataLayerId: z.string().nullable().default(null),
            projectId: z.string().nullable().default(null),
            siteId: z.string().nullable().default(null),
            teamId: z.string().nullable().default(null),
            theme: z.enum(["light", "dark"]),
        }),
        extension: z.object({
            id: z.string(),
            name: z.string(),
            slug: z.string(),
        }),
        surface: z.object({
            name: z.nativeEnum(Surfaces),
        }),
    }),
});
export const createUIExtensionHostInitializeEvent = (data) => ({
    type: UI_EXTENSION_HOST_INITIALIZE_EVENT_TYPE,
    data,
});
