// @netlify/sdk--ui-core/dist/message_schema/host/update_context.js
/*
 * The UPDATE_CONTEXT message is sent when the host has updated metadata (e.g. when routing state
 * has changed, when an authentication token has been revoked, etc.).
 *
 * The host must send the full context in this message, not a diff.
 */
import { z } from "zod";
export const UI_EXTENSION_HOST_UPDATE_CONTEXT_EVENT_TYPE = "NF::EUI_HOST::UPDATE_CONTEXT";
export const UIExtensionHostUpdateContextEventSchema = z.object({
    type: z.literal(UI_EXTENSION_HOST_UPDATE_CONTEXT_EVENT_TYPE),
    data: z.object({
        context: z.object({
            auth: z.object({
                netlifyToken: z.string().nullable().default(null),
                providerToken: z.string().nullable().default(null),
                providerTokens: z
                    .array(z.object({
                    externalId: z.string().min(1).trim(),
                    token: z.string().min(1).trim(),
                    connectionId: z.string().min(1).trim(),
                }))
                    .default(() => []),
            }),
            configurationId: z.string().nullable().default(null),
            dataLayerId: z.string().nullable().default(null),
            projectId: z.string().nullable().default(null),
            siteId: z.string().nullable().default(null),
            teamId: z.string().nullable().default(null),
            theme: z.enum(["light", "dark"]),
        }),
    }),
});
export const createUIExtensionHostUpdateContextEvent = (data) => ({
    type: UI_EXTENSION_HOST_UPDATE_CONTEXT_EVENT_TYPE,
    data,
});
