// @netlify/sdk--ui-core/dist/surfaces.js
/**
 * Surfaces is an enum containing the unique names of each surface (location in the Netlify UI) that a
 * UI extension can render to.
 *
 * The SDK exposes which surface your extension UI app is currently rendering to, so you can use this enum for
 * routing and display the correct UI for a given surface to a user.
 */
export var Surfaces;
(function (Surfaces) {
    Surfaces["TeamConfiguration"] = "extension-team-configuration";
    Surfaces["ConnectConfiguration"] = "extension-connect-configuration";
    Surfaces["SiteAccessConfiguration"] = "extension-site-access-configuration";
    Surfaces["SiteBuildDeployConfiguration"] = "extension-site-build-and-deploy-configuration";
    Surfaces["SiteGeneralConfiguration"] = "extension-site-general-configuration";
    Surfaces["SiteNotificationsConfiguration"] = "extension-site-notifications-configuration";
    Surfaces["SiteConfiguration"] = "extension-top-level-site-configuration";
    Surfaces["VisualEditorConfiguration"] = "extension-visual-editor-configuration";
})(Surfaces || (Surfaces = {}));
